pkgs/by-name/gh/ghostty/package.nix

```diff
@@ -1,153 +1,205 @@
-  # doCheck is set to false because unit tests currently fail inside the Nix sandbox.
-  doCheck = false;
-  doInstallCheck = true;
+  # Avoid using runtime hacks to help find X11
+  postPatch = lib.optionalString (appRuntime == "gtk") ''
+    substituteInPlace src/apprt/gtk/x11.zig \
+      --replace-warn 'std.DynLib.open("libX11.so");' 'std.DynLib.open("${lib.getLib libX11}/lib/libX11.so");'
+  '';
 
   deps = callPackage ./deps.nix {
     name = "${finalAttrs.pname}-cache-${finalAttrs.version}";
   };
 
+  strictDeps = true;
+
+  nativeBuildInputs =
+    [
+      ncurses
+      pandoc
+      pkg-config
+      removeReferencesTo
+      zig_hook
+    ]
+    ++ lib.optionals (appRuntime == "gtk") [
+      glib # Required for `glib-compile-schemas`
+      wrapGAppsHook4
+    ];
+
+  buildInputs =
+    [
+      glslang
+      oniguruma
+    ]
+    ++ lib.optional (appRuntime == "gtk" && withAdwaita) libadwaita
+    ++ lib.optional (appRuntime == "gtk") libX11
+    ++ lib.optional (renderer == "opengl") libGL
+    ++ lib.optionals (fontBackend == "fontconfig_freetype") [
+      bzip2
+      fontconfig
+      freetype
+      harfbuzz
+    ];
+
   zigBuildFlags =
     [
       "--system"
       "${finalAttrs.deps}"
       "-Dversion-string=${finalAttrs.version}"
+
+      "-Dapp-runtime=${appRuntime}"
+      "-Dfont-backend=${fontBackend}"
+      "-Dgtk-adwaita=${lib.boolToString withAdwaita}"
+      "-Drenderer=${renderer}"
     ]
     ++ lib.mapAttrsToList (name: package: "-fsys=${name} --search-prefix ${lib.getLib package}") {
       inherit glslang;
     };
 
   zigCheckFlags = finalAttrs.zigBuildFlags;
 
-  outputs = [
-    "out"
-    "terminfo"
-    "shell_integration"
-    "vim"
-  ];
+  # Unit tests currently fail inside the sandbox
+  doCheck = false;
 
-  postInstall = ''
-    mkdir -p "$terminfo/share"
-    mv "$out/share/terminfo" "$terminfo/share/terminfo"
-    ln -sf "$terminfo/share/terminfo" "$out/share/terminfo"
+  /**
+    Ghostty really likes all of it's resources to be in the same directory, so link them back after we split them
 
-    mkdir -p "$shell_integration"
-    mv "$out/share/ghostty/shell-integration" "$shell_integration/shell-integration"
-    ln -sf "$shell_integration/shell-integration" "$out/share/ghostty/shell-integration"
+    - https://github.com/ghostty-org/ghostty/blob/4b4d4062dfed7b37424c7210d1230242c709e990/src/os/resourcesdir.zig#L11-L52
+    - https://github.com/ghostty-org/ghostty/blob/4b4d4062dfed7b37424c7210d1230242c709e990/src/termio/Exec.zig#L745-L750
+    - https://github.com/ghostty-org/ghostty/blob/4b4d4062dfed7b37424c7210d1230242c709e990/src/termio/Exec.zig#L818-L834
+
+    terminfo and shell integration should also be installable on remote machines
+
+    ```nix
+    { pkgs, ... }: {
+      environment.systemPackages = [ pkgs.ghostty.terminfo ];
+
+      programs.bash = {
+        interactiveShellInit = ''
+          builtin source ${pkgs.ghostty.shell_integration}/bash/ghostty.bash
```

Comment:
Likely still want to wrap the example in a test on TERM since the integration normally only runs after checking `GHOSTTY_RESOURCES_DIR` exists.

```bash
if [[ "$TERM" == "xterm-ghostty" ]]; then
...
fi
```